app/routers/study_dashboard.py

```
@@ -0,0 +1,37 @@
+    학습 대시보드의 캘린더 학습 기록을 조회합니다.
+    - 해당 월의 모든 날짜에 대한 학습 기록 여부
+    - 학습한 날의 경우 관련 퀴즈 정보 포함
+    """
+    try:
+        user = request.state.user  # 미들웨어에서 설정된 사용자 정보
+        service = StudyDashboardService(db)
+        return await service.get_calendar_study_records(user.id, year, month)
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Failed to fetch calendar study records: {str(e)}"
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)